usecases/payload_parser/payload.go

```diff
@@ -151,74 +27,98 @@ var (
 	errIsInvalidDataType  = fmt.Errorf("invalid type")
 )
 
-func (p *Parser) ValidatePayload(table models.Table, json []byte) (map[models.FieldName]string, error) {
+func (p *Parser) ParsePayload(table models.Table, json []byte) (models.ClientObject, map[models.FieldName]string, error) {
 	if !gjson.ValidBytes(json) {
-		return nil, errIsInvalidJSON
+		return models.ClientObject{}, nil, errIsInvalidJSON
 	}
 
 	errors := make(map[models.FieldName]string)
+	out := make(map[string]any)
 	result := gjson.ParseBytes(json)
+
+	// Check fields that are always mandatory, regardless of the table definition
+	for _, name := range []string{"object_id", "updated_at"} {
+		value := result.Get(name)
+		if !value.Exists() || value.Type == gjson.Null {
+			errors[models.FieldName(name)] = errIsNotNullable.Error()
+		}
+		if name == "object_id" && result.String() == "" {
+			errors[models.FieldName(name)] = errIsNotNullable.Error()
+		}
+	}
+
 	for name, field := range table.Fields {
 		value := result.Get(string(name))
-		if !value.Exists() {
+		if !value.Exists() || value.Type == gjson.Null {
 			if !field.Nullable {
 				errors[name] = errIsNotNullable.Error()
 			}
+			out[string(name)] = nil
 			continue
 		}
 
-		validateField, ok := p.validators[field.DataType]
+		parseField, ok := p.parsers[field.DataType]
 		if !ok {
-			return nil, fmt.Errorf("%w: %s", errIsInvalidDataType, field.DataType.String())
+			return models.ClientObject{}, nil, fmt.Errorf("%w: %s",
+				errIsInvalidDataType, field.DataType.String())
 		}
-		if err := validateField(value); err != nil {
+		if val, err := parseField(value); err != nil {
 			errors[name] = err.Error()
+		} else {
+			out[string(name)] = val
 		}
 	}
 	if len(errors) > 0 {
-		return errors, nil
+		return models.ClientObject{}, errors, nil
 	}
-	return nil, nil
+	return models.ClientObject{
+		TableName: table.Name,
+		Data:      out,
+	}, nil, nil
 }
 
 func NewParser() *Parser {
-	validators := fieldValidator{
-		models.Timestamp: func(result gjson.Result) error {
-			_, err := time.Parse(time.RFC3339, result.String())
-			if err != nil {
-				return fmt.Errorf("%w: expected format YYYY-MM-DDThh:mm:ss[+optional decimals]Z, got %s", errIsInvalidTimestamp, result.String())
+	validators := fieldParser{
+		models.Timestamp: func(result gjson.Result) (any, error) {
+			t, err1 := time.Parse(time.RFC3339, result.String())
+			if err1 == nil {
+				return t.UTC(), nil
+			}
+			t, err2 := time.Parse("2006-01-02 15:04:05.9", result.String())
+			if err2 != nil {
+				return nil, fmt.Errorf("%w: expected format \"YYYY-MM-DD hh:mm:ss[+optional decimals]\" or \"YYYY-MM-DDThh:mm:ss[+optional decimals]Z\", got \"%s\"", errIsInvalidTimestamp, result.String())
 			}
-			return nil
+			return t.UTC(), nil
```

Comment:
I think this is clearer like this 
```go
if t, err := time.Parse(time.RFC3339, result.String()); err == nil {
      return t.UTC(), nil				
}
if t, err = time.Parse("2006-01-02 15:04:05.9", result.String()); err == nil {
      return t.UTC(), nil				
}
return nil, fmt.Errorf("%w: expected format \"YYYY-MM-DD hh:mm:ss[+optional decimals]\" or \"YYYY-MM-DDThh:mm:ss[+optional decimals]Z\", got \"%s\"", errIsInvalidTimestamp, result.String())
```

I thiknk the pattern "try to parse, if no error early return else cotinue and return error if no parsing works" is a good "common" pattern we could use for every parsers (still, since others are single case based, it's not really needed)

Reply:
true!